testing/internal-testing/src/main/groovy/org/gradle/util/GroovyNullMarked.groovy

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.util
+
+import javax.annotation.Nonnull
+import javax.annotation.meta.TypeQualifierDefault
+import java.lang.annotation.Documented
+import java.lang.annotation.ElementType
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.lang.annotation.Target
+
+@Target([ElementType.TYPE, ElementType.PACKAGE])
+@Nonnull
+@TypeQualifierDefault([ElementType.METHOD, ElementType.PARAMETER])
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@interface GroovyNullMarked {}
```

Comment:
Is this still necessary with the module-stripped version of JSpecify that was added later?

Reply:
Yes, the Groovy compiler is unhappy with type annotations in various ways 😢 

A very basic example:
```
Annotation @org.jspecify.annotations.Nullable is not allowed on element PARAMETER
       void setWarmups(@Nullable String warmups) {
```

I also witnessed Groovy compiler crash on more complex uses like when annotating array return types.

I would leave this out of scope of this PR that only migrate the production code, not test code.